### DISCUSSION TITLE
Playing multiple files outside setup()

### DISCUSSION THREAD

**Discussion by chuck808** [pschatzmann/arduino-audio-tools#551] [Q&A] [answered]
Hi,

At the risk of sounding completely stupid, I've been trying to play a couple of .mp3 files with a pause between them, but I can't seem to get it to work outside of the setup() 

I know it may be very bad form, but I think the easiest way to explain it to use a working example from another library that doesn't work for ESP32 :(

Using the TMRpcm library, I can call a file like this -

`tmrpcm.play((char*)"track1.wav");`

wait some period of time then - 

`tmrpcm.play((char*)"track2.wav");`

Then to ensure the audio file plays to the end - 

```
// Wait till finished talking
while (tmrpcm.isPlaying()) {
    delay(10);
 }
```
Any assistance would be greatly appreciated.

Regards

**Comment by pschatzmann** [comment, marked as answer]
Why don't you just use the AudioPlayer class?  See https://github.com/pschatzmann/arduino-audio-tools/wiki/The-Audio-Player-Class
The delay can be defined on the AudioSource with setTimeoutAutoNext()
